src/status_im/communities/core.cljs

```diff
@@ -780,3 +780,24 @@
                                 :community-id community-id
                                 :public-key   public-key
                                 :role-id      role-id})]})
+
+(rf/defn mute-chat-failed
+  {:events [:community/mute-community-failed]}
+  [{:keys [db]} community-id muted? error]
+  (log/error "mute community failed" community-id error)
+  {:db (assoc-in db [:communities community-id :muted] (not muted?))})
```

Comment:
yeah you are right should just be `muted` instead of `(not muted?)`

Reply:
actually, I think you should just remove this code and move the other  db change to on-success so the db is updated if the request is successful

Reply:
@J-Son89  Removing this code does not really go well with the flow of events. 
When you mute a community, the local db is updated first then the request is sent. The effect of this is  a real-time update of the ui that the community has been muted. 
Updating the ui after mute-community has been processed successfully takes time and sometime does not seem to work untill the app is restarted. Thats why we are updating the community to the previous state when the request fails. I think this code is okay as it is

Reply:
This topic surfaces every now and then about *what we should do if the RPC call fails*. My (imperfect) understanding is more or less like this:

- Many times, the mobile app does optimistic updates on the app db state, i.e. we call the RPC endpoint and simultaneously update the app db state. The `:on-success` of the RPC call is then used only if it returns new/unique data that wasn't available before the RPC call. The advantage of doing this is that it help make the app feel snappier.
- If we use an optimistic update AND the RPC call fails, then (ideally) we should rollback the app db changes in the `:on-error` event handler. We don't do this sometimes, we just log that something odd happened. This is bad in general. It is acceptable for now, but I suspect in the future we'll establish a stronger practice.
- We can avoid the optimistic update entirely as well, and some events do it, i.e. the app db state is only updated *after* a successful RPC call. This can cause a slightly longer delay for the UI to update.
- There's pretty much zero chance of the RPC call failing due to network conditions, because it's all local. So that's one big reason we treat RPC calls almost like function calls in Clojure. In other words, a failure always means something happened, but never network related. This is very different from re-frame apps that do a lot of remote network calls (often HTTP ones), where those apps need to be much more strict about network failures.

Reply:
> Updating the ui after mute-community has been processed successfully takes time and sometime does not seem to work untill the app is restarted. 

This though is concerning, it looks like there might be issues, it should not happen, so worth looking into.

The code is ok, we use this pattern somewhere, optimistically updating the community, and this endpoint does not provide an updated community response, so that's ok

Reply:
wow nice thanks for this explanation @ilmotta

Reply:
@cammellos can be looked into in a separate  issue?